src/Driver.cpp

```diff
@@ -619,6 +619,41 @@ pair<int, bool> Driver::extractPacketFromInternalBuffer(uint8_t* buffer, int out
     return make_pair(result_size, false);
 }
 
+int Driver::readRaw(uint8_t* buffer, int out_buffer_size)
+{
+    return readRaw(buffer, out_buffer_size, getReadTimeout());
+}
+
+int Driver::readRaw(uint8_t* buffer, int out_buffer_size, base::Time const& timeout)
```

Comment:
In principle, I think this is useful but the internal buffer management can be misleading. For me, it would make more sense if you didn't add the bytes in the internal buffer otherwise the user will read the same thing twice (in case the raw bytes contain an extractable packet). Generally, I think what is most likely to happen is that the user will read something with readRaw() and then, in the next call to readPacket() he/she will get something unexpected...

Reply:
It actually reads N bytes from the byte stream, regardless of whether these bytes were already in the internal buffer (from a previous call to e.g. readPacket) or not, so it's impossible to read the same bytes twice. I'll change the code to not passing through the internal buffer at all since the current code seems confusing.